### PR TITLE
QA issue fixes

### PR DESCRIPTION
**Pull request by jpuri**
The PR has fix for issues discovered during QA:

1. File fetch job not starting as preference for security alert is enabled
2. Returning JSON RPC error from infura in correct format

### CODE REVIEW

File: src/ppom-controller.ts [hunk 1]
```diff
@@ -771,24 +776,28 @@ export class PPOMController extends BaseControllerV2<
     method: string,
     params: Record<string, unknown>,
   ): Promise<any> {
-    return new Promise((resolve, reject) => {
+    return new Promise((resolve) => {
       // Throw error if number of request to provider from PPOM exceed the limit for current transaction
```
Comment (digiwand) [inline src/ppom-controller.ts:780]:
```suggestion
      // Resolve with error if number of PPOM requests to provider exceeds the limit for the current transaction
```

File: src/ppom-controller.ts [hunk 2]
```diff
@@ -771,24 +776,28 @@ export class PPOMController extends BaseControllerV2<
...
       if (this.#providerRequests > this.#providerRequestLimit) {
-        reject(PROVIDER_ERRORS.limitExceeded());
+        resolve(PROVIDER_ERRORS.limitExceeded());
         return;
       }
       this.#providerRequests += 1;
       // Throw error if the method called on provider by PPOM is not allowed for PPOM
```
Comment (digiwand) [inline src/ppom-controller.ts:786]:
```suggestion
      // Resolve with error if the provider method called by PPOM is not allowed for PPOM
```

File: src/util.ts [hunk 3]
```diff
@@ -20,7 +20,7 @@ export const PROVIDER_ERRORS = {
     jsonrpc: '2.0',
     id: IdGenerator(),
     error: {
-      code: -32005,
+      code: -32601,
```
Comment (digiwand) [inline src/util.ts:23]: why -32601? I see it represents methodNotFound in https://github.com/MetaMask/rpc-errors/blob/main/src/error-constants.ts#L11. Based on the message passed after this line, `Limit exceeded`, I would think the original `-32005` would be expected.

could we use constants for these values?

Comment (digiwand) [inline src/util.ts:23]: why `-32601`? I see it represents methodNotFound in https://github.com/MetaMask/rpc-errors/blob/main/src/error-constants.ts#L11. Based on the message passed after this line, `Limit exceeded`, I would think the original `-32005` would be expected.

could we use constants for these values?

Comment (digiwand) [inline src/util.ts:23]: I'm not understanding this change. why `-32601`? I see it represents methodNotFound in https://github.com/MetaMask/rpc-errors/blob/main/src/error-constants.ts#L11. Based on the message passed after this line, `Limit exceeded`, I would think the original `-32005` would be expected.

could we use constants for these values?

Comment (jpuri) [reply src/util.ts:23]: This value was suggested by Blockaid team

Comment (jpuri) [reply src/util.ts:23]: Good catch, I fixed this.

Comment (blackdevelopa) [reply src/util.ts]: > could we use constants for these value

Hey @jpuri any chance we can include this?

Comment (jpuri) [reply src/util.ts]: They are not going to be used at many places. Constants are not very useful I think.

**Review by digiwand — APPROVED** [review]
no blockers. I think it'd be nice to follow-up on https://github.com/MetaMask/ppom-validator/pull/39#discussion_r1294870061

**Pull request merged**